sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -470,6 +469,45 @@ public override int GetHashCode()
             return left is null ? right is null : left.CompareTo(right) >= 0;
         }
 
+        /// <summary>
+        /// Converts the string representation of a ResourceIdentifier to the equivalent <see cref="ResourceIdentifier"/> structure.
+        /// </summary>
+        /// <param name="input"> The id string to convert. </param>
+        /// <returns> A class that contains the value that was parsed. </returns>
+        /// <exception cref="FormatException"> when resourceId is not a valid <see cref="ResourceIdentifier"/> format. </exception>
+        /// <exception cref="ArgumentNullException"> when resourceId is null. </exception>
+        /// <exception cref="ArgumentException"> when resourceId is empty. </exception>
+        public static ResourceIdentifier Parse(string input)
+        {
+            var result = new ResourceIdentifier(input);
+            result.Parse();
+            return result;
+        }
+
+        /// <summary>
+        /// Converts the string representation of a ResourceIdentifier to the equivalent <see cref="ResourceIdentifier"/> structure.
+        /// </summary>
+        /// <param name="input"> The id string to convert. </param>
+        /// <param name="result">
+        /// The structure that will contain the parsed value.
+        /// If the method returns true result contains a valid ResourceIdentifier.
+        /// If the method returns false, result will be null.
+        /// </param>
+        /// <returns> True if the parse operation was successful; otherwise, false. </returns>
+        public static bool TryParse(string input, out ResourceIdentifier? result)
+        {
+            try
+            {
+                result = Parse(input);
+                return true;
+            }
+            catch
```

Comment:
If there is no way to avoid Parse throwing, we should probably catch the specific exception that it throws.